ui/app/models/database/connection.js

```diff
@@ -9,66 +9,134 @@ const AVAILABLE_PLUGIN_TYPES = [
     value: 'mongodb-database-plugin',
     displayName: 'MongoDB',
     fields: [
+      { attr: 'plugin_name' },
       { attr: 'name' },
+      { attr: 'connection_url' },
+      { attr: 'verify_connection' },
+      { attr: 'password_policy' },
+      { attr: 'username', group: 'pluginConfig', show: false },
+      { attr: 'password', group: 'pluginConfig', show: false },
+      { attr: 'write_concern', group: 'pluginConfig' },
+      { attr: 'username_template', group: 'pluginConfig' },
+      { attr: 'tls', group: 'pluginConfig', subgroup: 'TLS options' },
+      { attr: 'tls_ca', group: 'pluginConfig', subgroup: 'TLS options' },
+    ],
+  },
+  {
+    value: 'mssql-database-plugin',
+    displayName: 'MSSQL',
+    fields: [
       { attr: 'plugin_name' },
+      { attr: 'name' },
+      { attr: 'connection_url' },
+      { attr: 'verify_connection' },
       { attr: 'password_policy' },
-      { attr: 'username', group: 'pluginConfig' },
-      { attr: 'password', group: 'pluginConfig' },
-      { attr: 'connection_url', group: 'pluginConfig' },
-      { attr: 'write_concern' },
-      { attr: 'creation_statements' },
+      { attr: 'username', group: 'pluginConfig', show: false },
+      { attr: 'password', group: 'pluginConfig', show: false },
+      { attr: 'username_template', group: 'pluginConfig' },
+      { attr: 'max_open_connections', group: 'pluginConfig' },
+      { attr: 'max_idle_connections', group: 'pluginConfig' },
+      { attr: 'max_connection_lifetime', group: 'pluginConfig' },
     ],
   },
 ];
 
+/**
+ * fieldsToGroups helper fn
```

Comment:
any chance this would be a good candidate for a helper? e.g. could this be used elsewhere, or is this pretty specific to this model?

Reply:
Not exactly, since it's pretty specific to the shape of the object defined above, `AVAILABLE_PLUGIN_TYPES`. If we use that pattern in the future, we could definitely move it to a shared helper